src/x509_str.c

```diff
@@ -827,6 +1051,30 @@ WOLFSSL_X509_STORE* wolfSSL_X509_STORE_new(void)
     return NULL;
 }
 
+#ifdef OPENSSL_ALL
+static void X509StoreFreeObjList(WOLFSSL_X509_STORE* store,
+                  WOLF_STACK_OF(WOLFSSL_X509_OBJECT)* objs)
+{
+    int i;
+    WOLFSSL_X509_OBJECT *obj = NULL;
+    int cnt = store->numAdded;
+
+    i = wolfSSL_sk_X509_OBJECT_num(objs) - 1;
+    while (cnt > 0 && i > 0) {
+        /* The inner X509 is owned by somebody else, NULL out the reference */
+        obj = wolfSSL_sk_X509_OBJECT_value(objs, i);
+        if (obj != NULL) {
+            obj->type = 0;
+            obj->data.x509 = NULL;
+        }
+        cnt--;
+        i--;
+    }
```

Comment:
This could also be a `crl` so maybe better to use `obj->data.ptr`. Same effect just showing that we don't want to free anything the `WOLFSSL_X509_OBJECT` holds.

Reply:
In the future it would be nice to up ref what gets put into the object list so that we can just call `pop_free` but that is not necessary on this PR.

Reply:
Great catch on the CRLs. Agreed this whole thing could use a second pass at some point to take a close look at better using refcounts, but I think its large enough as is.